### PR TITLE
Rename upgrade record to system

### PR DESCRIPTION
**Pull request by ninegua**
The "upgrade" record is renamed to "system" so that it can fit more things unrelated to upgrades.

### CODE REVIEW

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -8783,17 +8783,17 @@ and main_actor as_opt mod_env ds fs up =
 
     (* Compile the declarations *)
     let ae2, decls_codeW = compile_decs_public env ae1 ds v2en
-      (Freevars.captured_vars (Freevars.upgrade up))
+      (Freevars.captured_vars (Freevars.system up))
```
Comment (ggreif) [inline src/codegen/compile.ml:8786]:
```suggestion
      Freevars.(captured_vars (system up))
```

**Comment by ninegua** [comment]
@crusso Yes the heartbeat PR will come after this one is merged!

**Review by crusso — COMMENTED** [review]
Merge at your leasure.

**Comment by ggreif** [comment]
I am merging this now. I assume this is okay, @ninegua?

**Pull request merged**